content/docs/guides/multi-target-exporter.md

```diff
@@ -0,0 +1,521 @@
+---
+title: Understanding and using the multi-target exporter pattern
+---
+
+# Understanding and using the multi-target exporter pattern
+
+This guide will introduce you to the multi-target exporter pattern. To achieve this we will:
+
+* describe the multi-target exporter pattern and why it is used,
+* run the [blackbox](https://github.com/prometheus/blackbox_exporter) exporter as an example of the pattern,
+* configure a custom query module for the blackbox exporter,
+* let the blackbox exporter run basic metric queries against the Prometheus [website](https://prometheus.io),
+* examine a popular pattern of configuring Prometheus to scrape exporters using relabeling.
+
+## The multi-target exporter pattern?
+
+By multi-target [exporter](/docs/instrumenting/exporters/) pattern we refer to a specific design, in which:
+
+* the exporter will get the target’s metrics via a network protocol.
+* the exporter does not have to run on the machine the metrics are taken from.
+* the exporter gets the targets and a query config string as parameters of Prometheus’ GET request.
+* the exporter subsequently starts the scrape after getting Prometheus’ GET requests and once it is done with scraping.
+* the exporter can query multiple targets.
+
+This pattern is only used for certain exporters, namely the [blackbox](https://github.com/prometheus/blackbox_exporter) and the [SNMP exporter](https://github.com/prometheus/snmp_exporter).
+
+The reason is that we either can’t run the exporter on the targets, e.g. network gear speaking SNMP, or that we are explicitly interested in the distance, e.g. latency and reachability of a website from a specific point outside of our network.
+
+## Running multi-target exporters
+
+Multi-target exporters are flexible regarding their environment and can be run in many ways. As regular programs, in containers, as background services, on baremetal, virtual machines. Because they are queried and do query over network they do need appropriate open ports. Otherwise they are frugal.
+
+Now let’s try it out for yourself!
+
+Use [Docker](https://www.docker.com/) to start a blackbox exporter container running by running this in a terminal:
+
+```bash
+sudo docker run -p 9115:9115 prom/blackbox-exporter
+```
+
+You should see a few log lines and if everything went well the last one should report `msg="Listening on address"` as seen here:
+
+```
+level=info ts=2018-10-17T15:41:35.4997596Z caller=main.go:324 msg="Listening on address" address=:9115
+```
+
+## Basic querying of multi-target exporters
+
+There are two ways of querying:
+
+1. Querying the exporter itself. It has its own metrics, usually available at `/metrics`.
+1. Querying the exporter to scrape another target. Usually available at a "descriptive" endpoint, e.g. `/probe`.
+
+You can manually try the first query type with curl in another terminal. If you don’t have curl, you can also copy the part between the `'` into a browser’s address bar:
```

Comment:
I think it's reasonably to presume that everyone has curl, otherwise use an `<a>` to link there directly.

Reply:
You are right, most people do have curl, but it is not preinstalled on many popular distributions.
A link is a very good idea, I will change that.